src/analyzer/schema-generator.ts

```diff
@@ -101,6 +101,10 @@ export async function generateSchema({ ssgMatchResult, ...fileBrowserOptions }:
 
     const models = _.concat<Model>(pageModels, dataModels, objectModels);
 
+    if (!models.length && !pagesDir && !dataDir) {
```

Comment:
if there's no data then we should return null

Reply:
not really needed.
The returned type can receive undefined for dataDir and pagesDir and empty array for models.